runtime/amplitude/src/xcm_config.rs

```diff
@@ -246,8 +266,77 @@ impl ShouldExecute for DenyReserveTransferToRelayChain {
 	}
 }
 
-pub type Barrier = AllowUnpaidExecutionFrom<Everything>;
+// We will allow for BRZ location from moonriver
+struct ReserveAssetDepositedMatcher1;
+impl ReserveAssetDepositedMatcher for ReserveAssetDepositedMatcher1 {
+	fn matches(&self, multi_asset: &MultiAsset) -> bool {
+		let expected_multiloc = BRZ_location();
+
+		match multi_asset {
+			MultiAsset { id: AssetId::Concrete(loc), .. } if loc == &expected_multiloc =>
+				return true,
+			_ => return false,
+		}
+	}
+}
+
+// The beneficiary. We will always expect a location of the form
+// MultiLocation: {
+//     parents: 1,
+//     interior: { X3: {
+//        Parachain(SenderParachainId),
+//        PalletInstance(xx), // will point to the automation pallet
+//        GeneralKey{ length: xx, data: []  }, // this will hold the reference number for Mykobo
+// },
+// Keep in mind that this location will be re-anchored
+
+// TODO modify with automation's pallet instance
+struct DepositAssetMatcher1;
+impl DepositAssetMatcher for DepositAssetMatcher1 {
+	fn matches<'a>(
+		&self,
+		assets: &'a MultiAssetFilter,
+		beneficiary: &'a MultiLocation,
+	) -> Option<(u8, &'a [u8])> {
+		if let (
+			Wild(AllCounted(1)),
+			MultiLocation {
+				parents: 0,
+				interior: X2(PalletInstance(99), GeneralKey { length, data }),
+			},
+		) = (assets, beneficiary)
+		{
+			Some((*length, &*data))
+		} else {
+			None
+		}
+	}
+}
+
+pub struct MatcherConfigAmplitude;
+
+impl MatcherConfig for MatcherConfigAmplitude {
+	fn get_matcher_pairs() -> Vec<MatcherPair> {
+		vec![
+			MatcherPair::new(
+				Box::new(ReserveAssetDepositedMatcher1),
+				Box::new(DepositAssetMatcher1),
+			),
+			// Additional matcher pairs to be defined in the future
+		]
+	}
+	fn get_incoming_parachain_id() -> u32 {
+		MOONRIVER_PARA_ID
```

Comment:
I see you've added a comment in line 325, about additional match pairs.
But  `get_incoming_parachain_id()` is specific to `MOONRIVER_PARA_ID`. Does it mean only tokens from _**Moonriver**_ are supported? 

If that's the intention, maybe rename the struct to `MoonRiverMatcherForAmplitude` or something, to make it obvious on the first read.